src/aliceVision/lensCorrectionProfile/lcp.hpp

```diff
@@ -0,0 +1,725 @@
+#pragma once
+
+#include <string>
+#include <vector>
+#include <sstream>
+
+enum class LCPCorrectionMode
+{
+    VIGNETTE,
+    DISTORTION,
+    CA
+};
+
+enum class LCPReadingState
+{
+    WaitSequence,
+    FillCommonAndCameraSettings,
+    FillCameraSettings,
+    FillGeometryModel,
+    FillVignetteModel,
+    FillChromaticGreenModel,
+    FillChromaticBlueGreenModel,
+    FillChromaticRedGreenModel,
+    WaitNewModel
+};
+
+inline std::ostream& operator<<(std::ostream& os, const LCPReadingState& state)
+{
+    if (state == LCPReadingState::WaitSequence)
+        os << "WaitSequence";
+    else if (state == LCPReadingState::FillCommonAndCameraSettings)
+        os << "FillCommonAndCameraSettings";
+    else if (state == LCPReadingState::FillCameraSettings)
+        os << "FillCameraSettings";
+    else if (state == LCPReadingState::FillGeometryModel)
+        os << "FillGeometryModel";
+    else if (state == LCPReadingState::FillVignetteModel)
+        os << "FillVignetteModel";
+    else if (state == LCPReadingState::FillChromaticGreenModel)
+        os << "FillChromaticGreenModel";
+    else if (state == LCPReadingState::FillChromaticBlueGreenModel)
+        os << "FillChromaticBlueGreenModel";
+    else if (state == LCPReadingState::FillChromaticRedGreenModel)
+        os << "FillChromaticRedGreenModel";
+    else if (state == LCPReadingState::WaitNewModel)
+        os << "WaitNewModel";
+
+    return os;
+}
+
+
+struct settingsInfo
+{
+    float FocalLength = 0.f;
+    float FocusDistance = 0.f;
+    float ApertureValue = 0.f;
+
+    void reset()
+    {
+        FocalLength = 0.f;
+        FocusDistance = 0.f;
+        ApertureValue = 0.f;
+    }
+};
+
+struct RectilinearModel
+{
+    int Version = -1;
+    float FocalLengthX = 0.f;
+    float FocalLengthY = 0.f;
+    float ImageXCenter = 0.5f;
+    float ImageYCenter = 0.5f;
+    float ResidualMeanError = 0.f;
+    float ResidualStandardDeviation = 0.f;
+    float RadialDistortParam1 = 0.f;
+    float RadialDistortParam2 = 0.f;
+    float RadialDistortParam3 = 0.f;
+    float TangentialDistortParam1 = 0.f;
+    float TangentialDistortParam2 = 0.f;
+    float ScaleFactor = 1.f;
+    bool isEmpty = true;
+
+    void reset()
+    {
+        Version = -1;
```

Comment:
Rewriting code like `*this = RectilinearModel()` would result in single source of truth for initial values.

Reply:
Same in several places below.